### PR TITLE
Compute and Network: Features and bug fixes for sprint49

### PR DESCRIPTION
**Pull request by anuchandy**
Following issues are addressed in this PR.

#1566
#1600
#1726
#1731
#1759
#1761
#1763
#1764
#1769
#1770
#1771
#1773
#1775
#1776
#1777
#1780
#1781
#1783

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anuchandy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (v-ancha). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/util/utils.js [hunk 1]
```diff
@@ -1073,11 +1073,16 @@ exports.uuidGen = function () {
   return uuid.v4();
 };
 
+exports.toLowerCaseAndRemoveSpace = function (str) {
+  if (str === null || str === undefined) {
```
Comment (yugangw-msft) [inline lib/util/utils.js]: will 'if (str)' also work?

File: lib/commands/arm/vm/vmProfile._js [hunk 2]
```diff
@@ -85,6 +89,11 @@ __.extend(VMProfile.prototype, {
       virtualMachine.tags = {};
     }
 
+    if (!utils.stringIsNullOrEmpty(this.params.vmSize)) { 
+	  var vmHardwareProfile = new VMHardwareProfile(this.cli, this.params);
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmProfile._js]: indent?

**Comment by anuchandy** [comment]
@yugangw-msft let me send same PR from a separate branch instead of dev, so that I don't have to hold on fixes for next release.

**Pull request closed without merging**